sparta/sparta/functional/RegisterBits.hpp

```diff
@@ -150,10 +158,22 @@ namespace sparta
          * If the original is pointing to its own memory, that will be copied
          */
         RegisterBits(const RegisterBits & orig) :
-            local_storage_(orig.local_storage_),
-            local_data_(orig.local_data_ == nullptr ? nullptr : local_storage_.data()),
-            remote_data_(orig.local_data_ == orig.remote_data_ ? local_data_ : orig.remote_data_)
-        {}
+            num_bytes_(orig.num_bytes_)
+        {
+            if(num_bytes_ <= local_storage_.size()) {
+                local_storage_ = orig.local_storage_;
+                local_data_ = orig.local_data_ == nullptr ? nullptr : local_storage_.data();
```

Comment:
When does this happen?

Reply:
Curious, when can `num_bytes_` be <= local storage size and the original still have its local_data_ pointer set to `nullptr`?

Reply:
But if it has > 0 bytes, will it ever be `nullptr`?

Reply:
Yes, it could be if it is initialized by a `const uint8` pointer. In that case, only `remote_data_` pointer is initialized. `local_data_` is `nullptr` until value is changed.